ListOfQuestions.md

```diff
@@ -14,4 +14,13 @@
 				(if	(empty?	in)
 						out
 						(recur	(rest	in)
-						(conj out (str	"Alice	is	"	(first	in)))))))
\ No newline at end of file
+						(conj out (str	"Alice	is	"	(first	in)))))))
+
+Week1:
+* Reader Macro
+? ->>
+
+* pattern match as in scala:
+case xs :: xxs =>
+
+* check emptyness in function args.
```

Comment:
There are various techniques for dealing with this.

* Explicit checking
* Pre- and post- conditions: http://blog.fogus.me/2009/12/21/clojures-pre-and-post/
* New clojure.spec library: http://clojure.org/guides/spec#_spec_ing_functions (available starting from Clojure 1.9.0)